src/components/Tabs.tsx

```diff
@@ -264,15 +287,41 @@ function TabImpl<V extends string>(props: TabImplProps<V>) {
   );
 }
 
-export function getTabStyles() {
-  return {
-    baseStyles: Css.df.aic.hPx(32).pyPx(6).px1.br4.smEm.outline0.gray700.add("width", "fit-content").cursorPointer.$,
-    activeStyles: Css.lightBlue700.bgLightBlue50.$,
-    disabledStyles: Css.gray400.cursorNotAllowed.$,
-    focusRingStyles: Css.bgLightBlue50.bshFocus.$,
-    hoverStyles: Css.gray700.bgGray100.$,
-    activeHoverStyles: Css.bgLightBlue200.lightBlue700.$,
-  };
+export function getTabStyles(type?: TabType) {
+  const verticalPaddingPx = 6;
+  const sharedBaseStyles = Css.df.aic
+    .hPx(32)
+    .pyPx(verticalPaddingPx)
+    .px1.outline0.gray700.add("width", "fit-content").cursorPointer;
+
+  switch (type) {
+    case "lineTab":
+      // Decrease the bottom padding by the same amount of the new border width to prevent text layout shift
+      const borderBottomWidthPx = 4;
+      const borderBottomStyles = Css.bb
+        .add("borderBottomWidth", `${borderBottomWidthPx}px`)
+        .pbPx(verticalPaddingPx - borderBottomWidthPx).$;
+
+      return {
+        baseStyles: sharedBaseStyles.sm.$,
+        activeStyles: Css.add(borderBottomStyles).bLightBlue700.smEm.gray900.$,
```

Comment:
concatenating styles with `.add(foo)` seemed nice/cute, but maybe not conventional? LMK if object spreads are preferred

Reply:
Seems fine!